pkg/matcher/cel.go

```diff
@@ -24,30 +24,66 @@ const (
 	changedFilesTags = "files."
 )
 
+func splitByQuotes(s string) []string {
+	customSplit := func(r rune) bool {
+		return r == '"' || r == '\''
+	}
+	// Split the string based on custom split function
+	return strings.FieldsFunc(s, customSplit)
+}
+
+// getBranchFromExpression function extracts the value associated with the provided branchKey.
+//
+// ex: Given an expression like target_branch == "main", getBranchFromExpression returns "main".
+func getBranchFromExpression(branchKey string) string {
+	parts := strings.Split(branchKey, "==")
+
+	var branch string
+	if len(parts) == 2 {
+		// Trim any leading or trailing white spaces from the value
+		value := strings.TrimSpace(parts[1])
+		sVal := splitByQuotes(value)
+		// Only valid values for example, are target_branch == "main", target_branch == 'main' or source_branch == "*" etc..
+		// Invalid formats include target_branch == ["main"] or target_branch == main.
+		if len(sVal) >= 1 {
+			branch = sVal[0]
+		}
+	}
+	return branch
+}
+
 func celEvaluate(ctx context.Context, expr string, event *info.Event, vcx provider.Interface) (ref.Val, error) {
 	eventTitle := event.PullRequestTitle
-	if event.TriggerTarget == triggertype.Push {
-		eventTitle = event.SHATitle
-		// For push event the target_branch & source_branch info coming from payload have refs/heads/
-		// but user may or mayn't provide refs/heads/ info while giving target_branch or source_branch in CEL expression
-		// ex:  pipelinesascode.tekton.dev/on-cel-expression: |
-		//        event == "push" && target_branch == "main" && "frontend/***".pathChanged()
-		// This logic will handle such case.
-		splittedValue := strings.Split(expr, "&&")
-		for i := range splittedValue {
-			if strings.Contains(splittedValue[i], "target_branch") {
-				if !strings.Contains(splittedValue[i], "refs/heads/") {
-					event.BaseBranch = strings.TrimPrefix(event.BaseBranch, "refs/heads/")
+	var (
+		baseBranchValue string
+		headBranchValue string
+	)
+	splitValue := strings.Split(expr, "&&")
+	for i := range splitValue {
+		if strings.HasPrefix(strings.TrimSpace(splitValue[i]), "target_branch") {
+			targetBranchValue := getBranchFromExpression(splitValue[i])
+			if ok := branchMatch(targetBranchValue, event.BaseBranch); ok {
+				baseBranchValue = targetBranchValue
+				if baseBranchValue == "" {
```

Comment:
if it is empty means something wrong happened in the previous function, then why we want to make i true by assigning the event value

Reply:
the intention was to keep the event value when target_branch and source_branch was not given
but the place where i have added condition is wrong i should have added outside the entire if block

After reviewing your comment, I rechecked the scenario when neither the target_branch nor the source_branch is provided. It became clear that the check should not be performed in this context.

 Therefore, I've removed it.